tests/parsing/ConfParserTest.py

```diff
@@ -200,3 +200,27 @@ def test_settings_override_warning(self):
                                             'already been defined in section '
                                             'name. The previous setting will '
                                             'be overridden.')
+
+
+class ConfParserTestParseSections(unittest.TestCase):
+    test_file = """
+    [Default]
+    ignore = aaa,bbb
+    ignore += ccc
+    """
+
+    def setUp(self):
+        self.tempdir = tempfile.gettempdir()
+        self.file = os.path.join(self.tempdir, '.coafile')
+        with open(self.file, 'w') as file:
+            file.write(self.test_file)
+        self.uut = ConfParser()
+        self.sections = self.uut.parse(self.file)
+
+    def tearDown(self):
+        os.remove(self.file)
+
+    def test_multiple_ignore_in_single_settings(self):
+        expected = 'aaa,bbb\nccc'
```

Comment:
>self.uut.sections['default'].contents['ignore'] should equal ['aaa', 'bbb', 'ccc']

hmm... the string for the particular keys seems to not have parsed yet, would you mind checking the contents of the final ignores field , **after** it is parsed into a list, whether it also separates elements by `\n` or just by `,`. If it doesn't (most likely the case) some additional machinery needs to be added, either:
1. By you in this patch itself for detecting the multiple `+=` and adding `,`s, so that your final string looks like `'aaa,bbb,ccc'` otherwise I am worried its gonna be parsed as just 2 items namely `['aaa', 'bbb\nccc']`

or

2. while parsing to also separate items by `\n`

The first one seems easier to implement as there can be cases where a `\n` is introduced to break lines in the config file even when the separation of elements is not required.

Or @jayvdb was talking about whether introducing at least one comma would allow it to be separated into a list appropriately. Not sure how that's implemented in code, but you might wanna check into that

Reply:
refer to `test_parse_empty_elem_strip_section` , which shows value splitting occurring by comma.  It also shows how to test that.  It could be that the testing code used in this new test case doesnt show the split field , because it might be re-joined transparently.

StringConvertor is a tricky piece of voodoo.

Reply:
Please follow `test_parse_empty_elem_strip_section`.  Yes, it is very possible that your code, and the even the test, is correct, and that the splitting happens somewhere else.

But your test code doesnt show that the splitting occurs correctly.  The test code in `test_parse_empty_elem_strip_section` does verify that the splitting occurs.

As `+=` is about joining items together, it is critical that the tests for multiple `+=` demostrate that the join logic is correct.

I suspect that the code is not correct (or at least it isnt ideal), because it is adding `\n` as the separator, when `,` is the required separator for lists, and `\n` is stripped from list items.

Please note that we already "have" `Setting` joining occurring for section inheritance.
e.g.
```
[python]
files = **.py
ignore = foo.py

[python.limited]
ignore += bar.py
```

However, I have encountered problems with using `+=` with section inheritance.  Maybe it also doesnt work correctly?  IIRC, we have an issue about it, but it hasnt been investigated yet.

Maybe an extra `,` fixes it; i.e. 

```
[python]
files = **.py
ignore = foo.py

[python.limited]
ignore += ,bar.py
```

If so, your PR here is correct, and the joining separator for `+=` is a different pre-existing bug.

Reply:
@jayvdb In section.py if I change this 
```
if self.__contains__(key, ignore_defaults=True) and allow_appending:
            val = self[key]
            val.value = str(val._value) + '\n' + setting._value
```
with output 
```
(Pdb) p self['ignore'].value
'tests/bearlib/languages/documentation/documentation_extraction_testdata/*.py,\ntests/collecting/collectors_test_dir/bears/incorrect_bear.py,'
```
to 
```
if self.__contains__(key, ignore_defaults=True) and allow_appending:
            val = self[key]
            val.value = str(val._value) + ',' + setting._value
```
Then`,` will indeed seperate the list
But, when the user puts a `,` in coafile it will look like this
```
',tests/bearlib/languages/documentation/documentation_extraction_testdata/*.py,,tests/collecting/collectors_test_dir/bears/incorrect_bear.py,'
```

Reply:
Please improve the test method first.

I do not want to be discussing strings with separators.

And it is not efficient to discuss possible future code changes on a PR.  Try it.  If the tests pass, push it; then we can see whatever you can see, and no chance of miscommunicating.

I want to be reviewing test methods which clearly show lists that are already separated correctly.  Test methods are what I look at first.  They are the most important part of your patch.  They should explain your changes.

After better tests are passing, then we can look closely at your code changes that were required to achieve the goal of this issue.

fwiw, changing `\n` to `,` would be a "breaking" change.  We would need to assess the impact of that breakage.  That assessment would investigate whether the current seperator `\n` was ever useful to anyone.  If it was useful, then the breaking change isnt acceptable, and we need to explore other ways of solving it.

But before finding the "best" way to solve it, find *any* way to solve it with all tests passing.